app/controllers/routes_controller.rb

```diff
@@ -39,10 +40,17 @@ def destroy
   private
 
   def route_params
-    params.require(:route).permit(:name, railway_station_ids: [])
+    params.require(:route).permit(railway_station_ids: [])
   end
 
   def set_route
     @route = Route.find(params[:id])
   end
+
+  def station_position(stations)
+    RailwayStationsRoute.where(route: @route).delete_all
+    stations.each do |station|
+      @route.railway_stations_routes.create!(railway_station_id: station, position: stations.find_index(station))
```

Comment:
What actually this hook **stations.find_index(station)**  should do ?

Reply:
Return index of element `station` in `stations`. And that will be an station position.